sa/sa.go

```diff
@@ -1335,9 +1408,65 @@ func (ssa *SQLStorageAuthority) GetOrder(ctx context.Context, req *sapb.OrderReq
 		order.Authorizations = append(order.Authorizations, authzID)
 	}
 
+	names, err := ssa.namesForOrder(*order.Id)
+	if err != nil {
+		return nil, err
+	}
+	// The requested names are stored reversed to improve indexing performance. We
```

Comment:
Question: there isn't an index on the name column, and we don't (at least in this PR) do a select using that column. Are we planning on doing this in the future at some point? Or can we just store these in their normal form?

Reply:
I stored the names in reversed order after @jsha suggested it for consistency with `issuedNames`. 

At the time he said: "I can't think of a strong reason to argue in favor of forward names, and reversedNames gives slightly nicer index locality for subdomains of the same domain."

I think I'd lean towards keeping them reversed to make it easier to exploit an index on the column down the road if we require it. It's tough to land DB migrations in prod so the future-proof approach might be better.

Reply:
Hm, I think future proofing makes sense, lack of indexes has definitely bit us down the road in the past. That said if that is why we're doing it we should probably just add the index now, I want to say it'll be faster to remove an index we don't need than add one we do.

Reply:
> That said if that is why we're doing it we should probably just add the index now, I want to say it'll be faster to remove an index we don't need than add one we do.

I like that argument. I'm :+1: to adding the index now.